python-threatexchange/threatexchange/extensions/text_tlsh/text_tlsh.py

```diff
@@ -42,7 +42,7 @@ def validate_signal_str(cls, signal_str: str) -> str:
     def hash_from_str(cls, text: str) -> str:
         hash_str = str(tlsh.hash(text.encode()))
         if hash_str == "TNULL":  # Likely too short
-            return ""
+            raise ValueError("text is too short")
```

Comment:
(blocking): Hmm, I'm not sure about this behavior change. The contract (which is undocumented in the docstring...) previously was that if your hash doesn't apply for the input (too short, etc), you just return empty string. We check for empty string in all the places where hash is called for this reason. Otherwise we'd need to catch exceptions anyways. If behavior is exceptional, then exceptions are a good choice. However, if your API will regularly need to deal with it, Exceptions are actually much slower to process than having a truthy return, or making the return optional. 

Otherwise, imagine if you have 3 applicable SignalTypes for a piece for a content, and one of them needs a minimum length.

We'd expect this:

```
$ threatexchange hash -I 'example text'
type_a blah
type_b blah

$ 
```
tlsh which needs 100 bytes doesn't get printed

Instead what we'd get is:

```
$ threatexchange hash -I 'example text'
type_a blah
type_b blah
<Big stacktrace>
ValueError
```

Reply:
I see what you're saying with the large stacktrace for API usage - that makes sense. :) 
However, I still don't think its actually good user behavior to just return an empty string because we haven't documented this anywhere. It seems buggy rather than a known and understood behavior.
Do you think it makes sense if we caught the ValueError, like we're doing for FileNotFound?